### DISCUSSION TITLE
Is it possible to do hybrid modelling in GAMA?

### DISCUSSION THREAD

**Discussion by pszps** [gama-platform/gama#52] [Q&A] [answered]
Is there any support in GAMA to implement hybrid models, similar to what you can do in AnyLogic (https://www.anylogic.com/)? I am in particular interested in the following pairings:

- System Dynamics (aggregate, based on differential equations) & Agent-Based (object oriented)
- Discrete Event (process oriented) & Agent-Based (object oriented))

If the answer is yes, are there any tutorials or sample models you could recommend?

**Comment by lesquoyb** [comment, marked as answer]
Hello,
GAMA is first a multi-agent modeling platform, so this behaviour would be the default one and the architecture you will have to follow for your model.
That being said, you can mix it with system dynamics by creating agents that act following differential equations for example. This is illustrated by the models you can find in the `Plugin models` under `Ordinary Differential Equations` in gama's embedded models:
![image](https://github.com/gama-platform/gama/assets/6374469/b7717726-e743-448b-8a5d-43ce456582a7)
If you have more precise questions about how it, maybe @tnguyenh can help you/provide some of his own examples.

For discrete events, as far as I know, there's nothing to handle it. I think that @benoitgaudou have been working on a plugin to bring it into gama, but I don't know if it's working yet
